spec/system/signup_spec.rb

```diff
@@ -125,7 +126,8 @@
 
       activate_account.click_activate_account
 
-      expect(page).to have_current_path("/new-topic?category_id=#{category.id}")
+      expect(page).to have_current_path("/c/#{category.id}")
+      expect(composer).to be_opened
```

Comment:
This logic here is essentially what we are testing for:

https://github.com/discourse/discourse/blob/423593fe22630f3e243ceb04a86f8875d1977675/app/assets/javascripts/discourse/app/routes/new-topic.js#L25-L32